frontend/packages/knative-plugin/src/topology/knative-topology-utils.ts

```diff
@@ -826,6 +838,34 @@ export const getSubscriptionTopologyEdgeItems = (
   return edges;
 };
 
+export const getKnSourceKafkaTopologyEdgeItems = (
+  kafkaSource: K8sResourceKind,
+  managedKafkaConnections,
+): EdgeModel[] => {
+  const { data } = managedKafkaConnections;
+  const edges = data.reduce((acc, managedKafkaConnection) => {
+    const mkcServiceAccountSecretName =
+      managedKafkaConnection?.spec?.credentials?.serviceAccountSecretName;
+    const kafkaSourcePasswordSecretKeyRefName =
+      kafkaSource.spec?.net?.sasl?.password?.secretKeyRef?.name;
+    const kafkaSourceUserSecretKeyRefName = kafkaSource.spec?.net?.sasl?.user?.secretKeyRef?.name;
+    if (
+      mkcServiceAccountSecretName === kafkaSourcePasswordSecretKeyRefName &&
```

Comment:
I think this check for connection can be simplified. 
Maybe use CR metadata instead?

Reply:
https://user-images.githubusercontent.com/981838/112825544-cc70eb80-9083-11eb-8570-d88985f6528d.png

Reply:
@wtrocki yeah this is something if we do, will definitely be neat in UI logic but it may not support the use case where we would like to show a connection between KS and KC even if the user created `Kafka source` via CLI with bootstrap server/secrets ( SASL plain) as they may not add the annotations and thus connections won't be shown UI.

@vikram-raj will be good to check if the value is there as well before comparison and bootstrap server matches

Reply:
Knowing this requirement I would have made those secrets less generic. Something like CloudKafkaConnectionSecret etc. I think we can leave that now